src/edgepi/adc/edgepi_adc.py

```
@@ -979,6 +987,136 @@ def set_config(
+            register_values[ADCReg.REG_MODE2.value]
+            & data_rate.value.op_mask
+        ) | data_rate.value.op_code
+
+        def create_read_command_tup(is_first: bool, mux_p: CH, mux_n: CH) -> tuple:
+            inpmux_register_value = generate_mux_opcode(ADCReg.REG_INPMUX, mux_p, mux_n).op_code
+            if is_first:
+                # update only data rate & input multiplexing (luckily they are right beside eachother)
+                start_addr = ADCReg.REG_MODE2.value
+                register_list = [mode2_register_value, inpmux_register_value]
+            else:
+                # Only write the register value 0x06 (INPMUX), which stores info
+                # about how the multiplexer should read from the ADC channels (input pins).
+                start_addr = ADCReg.REG_INPMUX.value
+                register_list = [inpmux_register_value]
+
+            write_reg_cmd = self.adc_ops.unsafe_write_register_command(
+                start_addr, register_list
+            )
+
+            # write config registers, start the adc's conversions, wait, then read registers
+            return (
+                write_reg_cmd + self.adc_ops.start_adc(ADCNum.ADC_1.value),
+                conversion_delay,
+                self.adc_ops.read_adc(ADCNum.ADC_1.value),
+            )
+
+        mux_pairs = (
+            [(channel, CH.AINCOM) for channel in channel_list] +
+            [(diff_mode.value.mux_p, diff_mode.value.mux_n) for diff_mode in differential_pairs]
+        )
+        # get instructions needed to send for each input
+        command_tup_list = [
+            create_read_command_tup(i == 0, mux_p, mux_n)
+            for i, (mux_p, mux_n) in enumerate(mux_pairs)
+        ]
+
+        # TODO: what happens if the SPI read fails (or something else fails in ADC during)?
+        # Register values will be unknown until we read from it again?
+        data_list = self.spi_apply_adc_commands(command_tup_list)
```

Do you have any suggestion for how I should handle the case where spi commands fail? Is it okay to let the exception bubble up & have `EdgePiADC.__state` not yet updated?

---

I'm happy with letting any exception bubble up, since the same will occur with `single_sample()`